src/main/java/frc/lib/team3061/drivetrain/Drivetrain.java

```diff
@@ -666,10 +667,192 @@ public boolean isMoveToPoseEnabled() {
     return this.isMoveToPoseEnabled;
   }
 
+  // method to convert swerve module number to location
+  private String getSwerveLocation(int swerveModuleNumber) {
+    switch (swerveModuleNumber) {
+      case 0:
+        return "FL";
+      case 1:
+        return "FR";
+      case 2:
+        return "BL";
+      case 3:
+        return "BR";
+      default:
+        return "UNKNOWN";
+    }
+  }
+
+  /*
+   * Checks the swerve module to see if it is moving as expected or is in the rotation threshold
+   * @param swerveModuleNumber the swerve module number to check
+   */
+
+  private void checkSwerveModule(
+      int swerveModuleNumber,
+      double angleTarget,
+      double angleTolerance,
+      double velocityTarget,
+      double velocityTolerance) {
+    // Need to check for both velocity and rotation
+
+    // Velocity check
+    if (Math.abs(inputs.swerve[swerveModuleNumber].driveVelocityMetersPerSec)
```

Comment:
As written, the test will pass if the wheel is aligned but spinning in the wrong direction. I recommend moving the velocity check into the if and else if below that knows if the wheels is aligned as expected or 180° offset. For the as expected alignment, the velocity can be checked as specified. For the 180° offset, the velocity can be negated and then checked. The Math.abs will no longer be needed.

I can help with this if you would like!